### DISCUSSION TITLE
How can string popup a char

### DISCUSSION THREAD

**Discussion by Decodetalkers** [slint-ui/slint#1372] [Q&A]
I want to push back a char from a string ,how to do that in slint?

**Comment by ogoffart** [comment]
in slint you can just use the `+`  or `+=` on string.
eg:

```
_ := Window := { 
   t := Text { text: "foo"; }
   TouchArea { clicked => { t.text += "0"; } }
}
```
([preview](https://slint-ui.com/releases/0.2.4/editor/?snippet=_+%3A%3D+Window+%7B+%0A+++t+%3A%3D+Text+%7B+text%3A+%22foo%22%3B+%7D%0A+++TouchArea+%7B+clicked+%3D%3E+%7B+t.text+%2B%3D+%220%22%3B+%7D+%7D%0A%7D%0A))

Or are you trying to do it in the progamming language with the SharedString api?

**Reply by Decodetalkers** [reply]
> in slint you can just use the `+` or `+=` on string. eg:
> 
> ```
> _ := Window := { 
>    t := Text { text: "foo"; }
>    TouchArea { clicked => { t.text += "0"; } }
> }
> ```
> 
> ([preview](https://slint-ui.com/releases/0.2.4/editor/?snippet=_+%3A%3D+Window+%7B+%0A+++t+%3A%3D+Text+%7B+text%3A+%22foo%22%3B+%7D%0A+++TouchArea+%7B+clicked+%3D%3E+%7B+t.text+%2B%3D+%220%22%3B+%7D+%7D%0A%7D%0A))
> 
> Or are you trying to do it in the progamming language with the SharedString api?

Sorry , I meen popup, is "abc" -> "ab", and I want to get the slice of the final char

**Reply by ogoffart** [reply]
This is currently not possible without using a bit of native code.

The way to achieve it with native code would be to register a callback and have the implementation pop the final char.

For example this will put the sting in uppercase: https://docs.rs/slint/latest/slint/docs/recipes/index.html#invoke-a-globally-registered-native-callback-from-slint   and you can do similar things that pop a char.
